functions.sh

```diff
@@ -156,10 +156,10 @@ substring_replace() {
 	_substring_replace_replacement="$1"
 	shift
 
-	_substring_replace_result=$(printf '%s' "$_substring_replace_str" | \
+	_substring_replace_result=$(printf '%s\n' "$_substring_replace_str" | \
 		sed -e "s!$_substring_replace_needle!$_substring_replace_replacement!g")
 
-	printf '%s\n' "$_substring_replace_result"
+	printf '%s' "$_substring_replace_result"
 }
```

Comment:
Can you explain to me why this change makes a difference?

Reply:
Minimal example on Solaris:

```
$ cat test
a
b
c
d
e
f
```

```
$ var=$(cat test) && printf '%s' "${var}" | sed -e 's/d/look-at-bottom-of-output/'
a
b
c
look-at-bottom-of-output
e
```

The `f` line was removed. In your case, `sed` was eating the last line of your `$contents` (which becomes `Makefile`) hence why all the targets were disappearing. Commenting out your `contents=$(replace ...)` lines one at a time helped to discover this.

Reply:
Fair enough.